python/sglang/srt/models/deepseek_v2.py

```diff
@@ -1368,6 +1353,33 @@ def __init__(
         self.logits_processor = LogitsProcessor(config)
         self.dp_size = get_attention_dp_size()
 
+    @staticmethod
+    def _initialize_shared_experts_fusion(config, tp_size) -> int:
+        n_share_experts_fusion = global_server_args_dict["n_share_experts_fusion"]
+
+        if (
+            global_server_args_dict.get("disable_shared_experts_fusion", False)
+            or config.architectures[0] != "DeepseekV3ForCausalLM"
+            or config.n_routed_experts != 256
+            or global_server_args_dict["enable_deepep_moe"]
+        ):
+            n_share_experts_fusion = 0
+            global_server_args_dict["n_share_experts_fusion"] = 0
+            logger.info(
+                "Only Deepseek V3/R1 tp mode can use shared experts fusion optimization. Shared experts fusion optimization is disabled."
+            )
+        elif n_share_experts_fusion is None:
+            n_share_experts_fusion = tp_size
+            global_server_args_dict["n_share_experts_fusion"] = tp_size
```

Comment:
nit: it seems this function is impure (indeed half pure and half impure), what about making it a pure function, i.e. do not do mutations inside it to reduce cognitive overhead